Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp

```diff
@@ -98,24 +89,21 @@ void NetworkMDNSRegister::closeAndForgetService(DNSServiceRef service)
     });
 }
 
-static void registerMDNSNameCallback(DNSServiceRef service, DNSRecordRef record, DNSServiceFlags, DNSServiceErrorType errorCode, void *context)
+static void registerMDNSNameCallback(DNSServiceRef service, DNSRecordRef record, DNSServiceFlags, DNSServiceErrorType errorCode, void* context)
 {
-    auto request = pendingRegistrationRequests().take(reinterpret_cast<uintptr_t>(context));
-    if (!request)
-        return;
+    auto request = std::unique_ptr<PendingRegistrationRequest>(reinterpret_cast<PendingRegistrationRequest*>(context));
```

Comment:
I suspect this would have sufficed:
```
std::unique_ptr request { reinterpret_cast<PendingRegistrationRequest*>(context) };
```

but ok.

Reply:
std::unique_ptr doesn't have deduction guides

Reply:
Sad :(

Reply:
The deduction guide is omitted because it wouldn't be able to tell the difference between the result of new T and new T[N]